idaes/core/util/testing.py

```diff
@@ -409,3 +414,31 @@ def get_reaction_rate_basis(b):
             return MaterialFlowBasis.mass
         else:
             return MaterialFlowBasis.other
+
+
+def _enable_scip_solver_for_testing(
+    name: str = "scip",
+) -> Union[Callable[[], None], None]:
+    ampl_module_scip, is_available = attempt_import("ampl_module_scip")
+    if not is_available:
+        _log.warning(
+            "ampl_module_scip must be installed to enable SCIP solver for testing"
+        )
+        return None
+
+    new_path_entry = ampl_module_scip.bin_dir
+    # TODO prepending new_path_entry to PATH means that the "testing" SCIP will "shadow"
+    # existing executable directories already on PATH
+    # this behavior would match the (implied) semantics of this function, i.e.
+    # "enable SCIP solver used for testing (at the expense of other non-testing SCIP executables)"
+    os.environ["PATH"] = os.pathsep.join([new_path_entry, os.environ["PATH"]])
+    Executable.rehash()
+
+    def undo_changes():
+        try:
+            os.environ["PATH"].remove(new_path_entry)
```

Comment:
Ughhh, I just realized that this won't work since `os.environ["PATH"]` is a `str` rather than a `list`. Serves me right for trying to be smart and implement a way to revert global changes to the environment...